TheMovieDB/TheMovieDB/MVP/Views/HomeScreenVC.swift

```diff
@@ -0,0 +1,140 @@
+//
+
+import Foundation
+import UIKit
+
+class HomeScreenVC: UIViewController {
+    
+    @IBOutlet weak var searchBar: UISearchBar!
+    
+    @IBOutlet weak var tableView: UITableView!
+    
+    var networkProvider: NetworkProvider = NetworkProvider()
+    
+    var filter: Filter = Filter()
+    
+    var movies: [Movie] = NetworkProvider().movies
+    
+    var selectedMovie: Movie?
+    
+    var tableCell: TableCell = TableCell()
+    
+    var collectionCell: CollectionCell = CollectionCell()
+    
+    let deviceIdiom = UIDevice.current.userInterfaceIdiom
+    
+    var collectionView: UICollectionView?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        let layout = UICollectionViewFlowLayout()
+        layout.scrollDirection = .vertical
+        layout.minimumLineSpacing = 1
+        layout.minimumInteritemSpacing = 1
+        layout.itemSize = CGSize(width: (view.frame.size.width/3)-4, height: (view.frame.size.height/3)-4)
+        
+        collectionView = UICollectionView(frame: .zero, collectionViewLayout: layout)
+        guard let collectionView = collectionView else {
+            return
+        }
+        collectionView.register(CollectionCell.self, forCellWithReuseIdentifier: CollectionCell.identifier)
+        collectionView.dataSource = self
+        collectionView.delegate = self
+        view.addSubview(collectionView)
+        collectionView.frame = view.bounds
+        
+        searchBar.searchTextField.textColor = UIColor.green
+        searchBar.searchTextField.backgroundColor = UIColor.black
+        searchBar.searchTextField.leftView?.tintColor = UIColor.green
+        searchBar.subviews[0].tintColor = UIColor.green
+        
+        switch deviceIdiom {
+        case .pad:
+            tableView.isHidden = true
+            collectionView.isHidden = false
+        default:
+            tableView.isHidden = false
+            collectionView.isHidden = true
+        }
+        
+        networkProvider.getMovies(completion: { [weak self] result in self?.movies = result
+            
+            DispatchQueue.main.async { [weak self] in self?.tableView.reloadData()
+            }
+        })
+    }
+}
+
+extension HomeScreenVC: UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return movies.count
+    }
+    
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let movie = movies[indexPath.row]
+        guard let cell = collectionView.dequeueReusableCell(withReuseIdentifier: CollectionCell.identifier, for: indexPath) as? CollectionCell else {
+            return UICollectionViewCell()
+        }
+        
+        collectionCell.setMovieCell(movie: movie, cell: cell)
+        return cell
+    }
+}
+
+
+extension HomeScreenVC: UICollectionViewDelegate {
+    
+}
+
+/*
```

Comment:
if you have pending task you should mark them with a `// TODO:` or a `// Note:` but you should never leave comments alone in the code